src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/BurdenEstimates/BurdenEstimateUploadController.kt

```diff
@@ -28,6 +29,31 @@ class BurdenEstimateUploadController(context: ActionContext,
             RepositoriesBurdenEstimateLogic(repos.modellingGroup, repos.burdenEstimates, repos.expectations),
             repos.burdenEstimates)
 
+    fun getUploadToken(): String
+    {
+        val path = getValidResponsibilityPath(context, estimateRepository)
+        val setId = context.params(":set-id").toInt()
+
+        // Check that this is a central estimate set
+        val metadata = estimateRepository.getBurdenEstimateSet(path.groupId,
+                path.touchstoneVersionId,
+                path.scenarioId,
+                setId)
+
+        if (metadata.isStochastic())
+        {
+            throw BadRequest("Stochastic estimate upload not supported")
+        }
+
```

Comment:
Should there be any validation checks on the filename? Is this the filename that's going to be used on the server, or is it just the filename on the client? If the former, is that a decision the client should be making? What if there's a clash of filenames?

Reply:
This filename won't be used on the server, other than we will check that the filename passed when uploading matches the original filename the token was fetched for. Tbh not sure how important this is, maybe the user should just be able to request an upload token, filename agnostic, and then use it in a future request with any file?

The client can pass any filename they want, we'll actually save the file using the uid encoded in the token claim (https://github.com/vimc/montagu-api/pull/360/files#diff-142e2f35e47563007a9f8cd358d540ebR50). Tbh I guess the filename is super irrelevant, the upload token (which the client is required to pass with each chunk POST) uniquely identifies the file. Maybe we just forget the filename altogether, would make things simpler?

The only reason we might care about filename is because modellers have actually requested that we save the filenames and return them as burden estimate set metadata, for their own very basic sanity checks on what they uploaded. But for that we only have to record the filename when the upload the file, not when requesting a token.

Reply:
Oh sorry I remembered my original thinking - we don't want anyone to use the same token twice! Including the filename *somewhat* guards against this. But maybe a better way would be to store the tokens in a db (the way we do for onetime tokens) and remove them after a file has been successfully uploaded.